src/shell/shell_misc.cpp

```diff
@@ -589,6 +592,8 @@ bool DOS_Shell::Execute(char * name,char * args) {
 		CALLBACK_RunRealInt(0x21);
 		/* Restore CS:IP and the stack */
 		reg_sp+=0x200;
+		if (result_errorcode)
+			dos.return_code = result_errorcode;
```

Comment:
In the CHOICE call, presumably it updates `dos.return_code`, which we're saving in `result_errorcode`:
``` c++
result_errorcode = dos.return_code;
```

But then here in shell_misc, we assign to `dos.return_code` again? 

I guess `dos.return_code` was cleared somewhere, and we need to re-assign it?

Is it possible to use any other struct member for this instead of a new global?

Maybe `dos.errorcode`? (just a question.. I don't know the answer)

Reply:
@kcgen The DOS error code is not properly returned when CHOICE becomes a program because it is indeed cleared/reset somewhere, so I had to re-assign it. This is why it took me some time to find such a workaround, or that the DOS return code is always 0 for CHOICE command (which can be tested with IF ERRORLEVEL command). The temporary return code is not used anywhere else and I am not quite sure if it really belongs to the dos struct member (dos.errorcode seems to suggest that it actually retains the error code at all times, but it is only supposed to be workaround for this particular command).